include/internal/refcount.h

```diff
@@ -16,26 +16,32 @@
 #  endif
 # endif
 
+# ifndef CRYPTO_REF_TYPE
+#  define CRYPTO_REF_TYPE int
+# endif
+
 # if defined(__STDC_VERSION__) && __STDC_VERSION__ >= 201112L \
      && !defined(__STDC_NO_ATOMICS__)
-# include <stdatomic.h>
-# define HAVE_C11_ATOMICS
+#  include <stdatomic.h>
+#  define HAVE_C11_ATOMICS
 # endif
 
 # if defined(HAVE_C11_ATOMICS) && defined(ATOMIC_INT_LOCK_FREE) \
      && ATOMIC_INT_LOCK_FREE > 0
 
-# define HAVE_ATOMICS 1
+#  define HAVE_ATOMICS 1
 
-typedef _Atomic int CRYPTO_REF_COUNT;
+typedef _Atomic CRYPTO_REF_TYPE CRYPTO_REF_COUNT;
```

Comment:
Doesn't this mean that we want to check ATOMIC_LONG_LOCK_FREE or ATOMIC_INT_LOCK_FREE depending on the CRYPTO_REF_TYPE?